repositories/src/test/scala/cromwell/pipeline/datastorage/dao/repository/ProjectConfigurationRepositoryTestImpl.scala

```diff
@@ -0,0 +1,39 @@
+package cromwell.pipeline.datastorage.dao.repository
+import cromwell.pipeline.datastorage.dto.{ ProjectConfiguration, ProjectConfigurationId, ProjectId }
+
+import scala.collection.mutable
+import scala.concurrent.Future
+
+class ProjectConfigurationRepositoryTestImpl extends ProjectConfigurationRepository {
+
+  private val configurations: mutable.Map[ProjectConfigurationId, ProjectConfiguration] = mutable.Map.empty
+
+  def addConfiguration(projectConfiguration: ProjectConfiguration): Future[Unit] = {
+    configurations += (projectConfiguration.id -> projectConfiguration)
+    Future.successful(())
+  }
+
+  def updateConfiguration(projectConfiguration: ProjectConfiguration): Future[Unit] = {
+    if (configurations.contains(projectConfiguration.id)) {
+      configurations += (projectConfiguration.id -> projectConfiguration)
+    }
+    Future.successful(())
+  }
+
+  def getById(id: ProjectConfigurationId): Future[Option[ProjectConfiguration]] =
+    Future.successful(configurations.get(id))
+
+  def getAllByProjectId(projectId: ProjectId): Future[Seq[ProjectConfiguration]] =
+    Future.successful(configurations.values.filter(_.projectId == projectId).toSeq)
+
+}
+
+object ProjectConfigurationRepositoryTestImpl {
+
+  def apply(configurationsSet: ProjectConfiguration*): ProjectConfigurationRepositoryTestImpl = {
```

Comment:
Is it ok to call something smthSet if it isn't set? Everything else is ok for me

Reply:
I called this a set because we can pass as many parameters as we like, provided that they have different id.

Reply:
I think this naming is a bit confusing. May be you should name it "configurations"? It shows that there are several of it and its type. What do you think?

Reply:
And it's valid for usersSet, runsSet and projectsSet too

Reply:
> I think this naming is a bit confusing. May be you should name it "configurations"? It shows that there are several of them and its type. What do you think?

 maybe you are right, so I'll change it

Reply:
@gominasian done